src/H5CX.c

```diff
@@ -1023,6 +1025,8 @@ H5CX_restore_state(const H5CX_state_t *api_state)
 
     /* Restore the VOL wrapper context */
     (*head)->ctx.vol_wrap_ctx = api_state->vol_wrap_ctx;
+    if (NULL != (*head)->ctx.vol_wrap_ctx)
```

Comment:
This is an actual bug fix.   (Might be the only actual code change of significance in this entire PR. ;-) )

Reply:
I feel like mixing real bugfixes should be separated from giant noise-reduction PRs.

Reply:
> I feel like real bugfixes should be separated from giant noise-reduction PRs.

Yes, I agree and try to avoid it also.  I can remove it, if you think it's enough of a distraction.   (It only affects code that's for async operations, so won't be an issue to port back to versions before 1.10, if the overall patch doesn't get pushed back)